clusterloader2/pkg/measurement/common/scheduler_latency.go

```diff
@@ -134,37 +146,91 @@ func (*schedulerLatencyMeasurement) String() string {
 	return schedulerLatencyMetricName
 }
 
-func (s *schedulerLatencyMeasurement) resetSchedulerMetrics(c clientset.Interface, host, provider, masterName string, masterRegistered bool) error {
-	_, err := s.sendRequestToScheduler(c, "DELETE", host, provider, masterName, masterRegistered)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
 // Retrieves scheduler latency metrics.
 func (s *schedulerLatencyMeasurement) getSchedulingLatency(c clientset.Interface, host, provider, masterName string, masterRegistered bool) ([]measurement.Summary, error) {
 	result := schedulingMetrics{
 		FrameworkExtensionPointDuration: make(map[string]*measurementutil.LatencyMetric),
 	}
-	data, err := s.sendRequestToScheduler(c, "GET", host, provider, masterName, masterRegistered)
+	for _, ePoint := range extentionsPoints {
+		result.FrameworkExtensionPointDuration[ePoint] = &measurementutil.LatencyMetric{}
+	}
+
+	schedulerMetrics, err := s.getSchedulingMetrics(c, host, provider, masterName, masterRegistered)
 	if err != nil {
 		return nil, err
 	}
 
-	samples, err := measurementutil.ExtractMetricSamples(data)
+	if s.initialLatency == nil {
+		return nil, fmt.Errorf("initial values are not available")
+	}
+	for k := range schedulerMetrics.preemptionEvaluationHist.Buckets {
```

Comment:
Please introduce a helper function to divide two histograms and use it for all below.

Reply:
added helper function to consolidate the substractions